pkg/utils/foreignCluster/getForeignCluster.go

```diff
@@ -71,11 +72,7 @@ func GetOlderForeignCluster(
 }
 
 // getAuthAddress retrieves the external address where the Authentication Service is reachable from the external world.
-func getAuthAddress(ctx context.Context, cl client.Client, authServiceAddress, namespace string) (string, error) {
-	if authServiceAddress != "" {
-		return authServiceAddress, nil
-	}
-
+func getAuthAddress(ctx context.Context, cl client.Client, namespace string) (string, error) {
```

Comment:
Nit. These two functions could now be refactored to retrieve the service only once in the `GetHomeAuthURL` function, and propagate it here and in `getAuthPort`.